plugins/auth-backend/src/providers/auth0/strategy.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import OAuth2Strategy from 'passport-oauth2';
+
+export interface Auth0StrategyOptionsWithRequest {
+    clientID: string;
+    clientSecret: string;
+    callbackURL: string;
+    domain: string;
+    passReqToCallback: true;
+}
+
+export default class Auth0Strategy extends OAuth2Strategy {
+    constructor(options: Auth0StrategyOptionsWithRequest, verify: OAuth2Strategy.VerifyFunctionWithRequest) {
+        const optionsWithURLs = {
+            ...options,
+            authorizationURL: `https://${options.domain}/authorize`,
+            tokenURL: `https://${options.domain}/oauth/token`,
+            userInfoURL: `https://${options.domain}/userinfo`,
+            apiUrl: `https://${options.domain}/api`,
+        }
+        super(optionsWithURLs, verify)
+    }
+
+    tokenParams(): object {
+        return {
+            'type': 'web_server',
+            'grant_type': 'client_credentials'
```

Comment:
This confuses me, I assume we would've wanted authorization_code?

Reply:
Yeah I assumed the same but they get the access token with that grant type 🤔 Example in their lib: https://github.com/auth0/passport-auth0/blob/096f789bea36a45a18d1a06accdd73decfb65131/lib/index.js#L143

Reply:
Looks like it's hardcoded to authorization_code in the base strategy, so you could just leave it out here: https://github.com/jaredhanson/passport-oauth2/blob/e20f26aad60ed54f0e7952928cbb64979ef8da2b/lib/strategy.js#L167

Reply:
Yes, it makes sense. I can test with authorization code to see if behaves according to the RFC

Reply:
It works as expected. Wonder why their lib has `client_credentials` 🤷